### PR TITLE
Add southbound API to pass notification to RackHD (refer to ODR682)

### PR DESCRIPTION
**Pull request by cgx027**
This is the PR to support https://github.com/RackHD/on-http/pull/377

### CODE REVIEW

**Comment by yyscamper** [comment]
@tannoa2 This is the notification API that you want.

**Comment by iceiilin** [comment]
looks great to me

File: lib/protocol/events.js [hunk 1]
```diff
@@ -140,6 +142,33 @@ function eventsProtocolFactory (
         );
     };
 
+    EventsProtocol.prototype.publishTaskNotification = function (nodeId, taskId, graphId, type, data) {
```
Comment (yyscamper) [inline lib/protocol/events.js]: Please add some JSDoc comment to tell the data can be either String or Object

**Comment by cgx027** [comment]
Remove WIP flag

**Comment by yyscamper** [comment]
:+1:

**Comment by WangWinson** [comment]
:+1:

**Pull request merged**